### PR TITLE
[G2M] Plotly - implementing unit tests

### PR DESCRIPTION
**Pull request by Hamaad102**
- Separated code into util file to allow for reusability and readability
- Implemented unit testing for functions necessary to generate Plotly graphs
Screenshot 2022-10-03 085029

### CODE REVIEW

**Comment by woozyking** [comment]
nice, let's get this also incorporate in the CI workflow

**Review by woozyking — APPROVED** [review]
let's try to keep the same convention of the commit message `<category> - <subject>`

File: src/components/plotly/pie/index.js [hunk 1]
```diff
@@ -21,22 +16,23 @@ const Pie = ({
...
-        textinfo: textinfo ?? getTextInfo({ showPercentage, showLabelName }),
-        hole: hole ?? (donut ? 0.4 : 0),
-      },
-      formatData,
-      ...props,
-    })}
-    {...props}
-  />
-)
+}) => {
```
Comment (geoerika) [inline src/components/plotly/pie/index.js:19]: @Hamaad102 @woozyking why was this change necessary? We usually use this short form of returning.

Comment (Hamaad102) [reply src/components/plotly/pie/index.js:19]: Sorry, this change wasn't necessary. I did this in the midst of figuring out what the values being sent to each component were.

**Review by kc-leung — COMMENTED** [review]
so it is currently only testing pyramid-bar ? there are some util code in use-transformed-data, can those be unit tested as well?

**Comment by Hamaad102** [comment]
@kc-leung it's testing the bar, pie and pyramid. The reason pyramid has imported data is because the functions it relies on needs unmanipulated data. The other two work with data that has gone through the TransformedData function. Line and scatter don't rely on anything external besides use pyramid.

File: src/components/plotly/pie/index.js (hunk 1 above)
Comment (geoerika) [reply src/components/plotly/pie/index.js:19]: No worries, I can fix it in my next PR, as I have some local changes already.
